File: solutions/cpp/pangram/2/pangram.cpp
#include "pangram.h"

namespace pangram {
    bool is_pangram(std::string string) {
        std::transform(string.begin(), string.end(), string.begin(), 
            [](const char& character) -> char {
                return std::tolower(character);
            });
        
        std::bitset<26> occurrences;
        
        for(const auto& character : string) {
            if(character >= 'a' && character <= 'z')
                occurrences.set(character - 'a');
        }

        return occurrences.all();
    }
}

File: solutions/haskell/pangram/1/src/Pangram.hs
module Pangram (isPangram) where
import Data.Char (toLower)

isPangram :: String -> Bool
isPangram s = all(\c -> countc c (map toLower s) > 0) alphabet
  where countc c = foldr (\cs acc -> if cs == c then acc + 1 else acc) 0
        alphabet = "abcdefghijklmnopqrstuvwxyz"